pkg/cmd/run/download/zip.go

```diff
@@ -54,7 +54,22 @@ func extractZipFile(zf *zip.File, dest safepaths.Absolute) (extractErr error) {
 	}
 
 	var df *os.File
-	if df, extractErr = os.OpenFile(dest.String(), os.O_WRONLY|os.O_CREATE|os.O_EXCL, getPerm(zm)); extractErr != nil {
+	flag := os.O_WRONLY | os.O_CREATE
+	if !force {
+		flag = flag | os.O_EXCL
+	}
+
+	if df, extractErr = os.OpenFile(dest.String(), flag, getPerm(zm)); extractErr != nil {
+		errExist := errors.Is(extractErr, os.ErrExist)
+		if errExist && skip {
+			return nil
+		}
+		if errExist {
+			return fmt.Errorf(
+				"%s already exists ((use `--clobber` to override or `--skip-existing` to skip))",
+				dest,
+			)
+		}
```

Comment:
Comparing this against `gh release download` logic:

https://github.com/cli/cli/blob/0acd40418a7e32e72f373d8b846a7bfadc110cfb/pkg/cmd/release/download/download.go#L343-L370